tools/lib/template_parser.py

```diff
@@ -349,6 +349,28 @@ def f(end_token: Token) -> None:
     if state.depth != 0:
         raise TemplateParserException("Missing end tag")
 
+    min_row: Optional[int] = None
+    for token in tokens:
+        if token.kind in ("handlebars_singleton_end", "html_singleton_end"):
+            continue
+        if token.tag in ("a", "br", "i", "span", "tbody", "textarea"):
```

Comment:
Maybe add a brief comment with some of the thinking from the commit message about how some of these, like textarea, may need the legacy style for whitespace reasons, and which others are likely just legacy things we should fix?

Reply:
I added some comments, and I also extracted the function.  Next time we're on a call I will give you a bit more context on this.  There is probably a better long term strategy for this code, which may be just some incremental tweaking but could also be more of a re-think on long line-wrapped opening tags.